gnuradio-runtime/lib/thread/thread_group.cc

```diff
@@ -72,21 +64,16 @@ void thread_group::remove_thread(boost::thread* thrd)
 void thread_group::join_all()
 {
     boost::shared_lock<boost::shared_mutex> guard(m_mutex);
-    for (std::list<boost::thread*>::iterator it = m_threads.begin();
-         it != m_threads.end();
-         ++it) {
-        (*it)->join();
+    for (auto& thrd : m_threads) {
```

Comment:
This is a memory-management-unrelated change, but honestly, with the change in data structure, keeping the old for-loop would've been stupid, :+1:

Reply:
Yeah the loop needed a minor change because of the type change anyway, otherwise I wouldn't have touched it here.